mistralrs-core/src/pipeline/mod.rs

```diff
@@ -363,11 +366,45 @@ pub trait NormalModel {
     }
 }
 
-struct InputMetadata {
-    input: Tensor,
-    positions: Vec<usize>,
-    positions_kernel: Tensor, // [bs, seq len]
-    context_lens: Vec<usize>,
+#[derive(Debug)]
+pub struct InputMetadata {
+    pub input: Tensor,
+    pub positions: Vec<usize>,
+    pub positions_kernel: Tensor, // [bs, seq len]
+    pub context_lens: Vec<usize>,
+}
+
+fn calculate_inputs_prompt_batched(
+    seq: &mut Sequence,
+    device: &Device,
+    chunk_size: usize,
+) -> Result<Vec<InputMetadata>> {
```

Comment:
Is it worth it making `InputMetadata` public? Or should this function return `ModelInputs`?